packages/calcite-components/src/components/filter/filter.tsx

```diff
@@ -167,9 +167,7 @@ export class Filter
 
   async componentWillLoad(): Promise<void> {
     setUpLoadableComponent(this);
-    if (this.items.length) {
-      this.updateFiltered(filter(this.items, this.value, this.filterProps));
-    }
+    this.updateFiltered(filter(this.items ?? [], this.value, this.filterProps));
```

Comment:
Should we check for `value` presence before triggering the filter change event? Emitting an event on load when there are no items nor filter value (noop) seems odd to me.

Reply:
@jcfranco the problem with this is that when the user clears the filter, the value will be an empty string but we do want to fire the event in this scenario.

I could check to make sure filter is not null or undefined. However, the default filter component value is an empty string.

Reply:
I don't think it would emit on load. The `updateFiltered` call on `componentWillLoad` doesn't emit.

Reply:
Oh, I see. It's because the [`emit` argument is falsy](https://github.com/Esri/calcite-design-system/blob/1cb3d9055f0f4d36effd83c1543e1baae78c275f/packages/calcite-components/src/components/filter/filter.tsx#L265) when there is no filter value. Ignore my comment then. Thanks for checking!